config.org
#+TITLE:     Doom config
#+AUTHOR:    Athul Suresh
#+EMAIL:     athulsure@gmail.com
#+STARTUP:   overview


* General
There are some settings where the values need to be different between my different laptops. Using the =system-name= is the easiest solution I've found to differentiate the machines.
#+begin_src elisp :tangle config.el

(setq mac-work "Athuls-MacBook-Pro.local")
(setq mac-personal "MtDoom.local")

#+end_src
* UI
** Font
I've experimented with multiple fonts in the past and nothing seemed good enough. That was until I found [[https://www.jetbrains.com/lp/mono/][JetBrains Mono]]. I've been using it ever since I saw the announcement on HN and I have it configured on all my dev environments. The one I've configured here is a patched version provided by [[https://www.nerdfonts.com/][Nerd Fonts]].
#+begin_src elisp :tangle config.el
;; Doom exposes five (optional) variables for controlling fonts in Doom. Here
;; are the three important ones:
;;
;; + `doom-font'
;; + `doom-variable-pitch-font'
;; + `doom-big-font' -- used for `doom-big-font-mode'; use this for
;;   presentations or streaming.
;;
;; They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
;; font string. You generally only need these two:
(setq doom-font (font-spec :family "JetBrainsMono Nerd Font" :size 15))
#+end_src

** Theme
I generally run my display at the lowest brightness setting since it's easier on the eye when I have to stare at it for long stretches of time. Most of the dark themes are barely visible at low brightness, and so I've switched to using light themes.
#+begin_src elisp :tangle config.el
;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:
(setq doom-theme 'doom-one-light)
#+end_src
*** TODO Spend some time trying out [[https://github.com/protesilaos/modus-themes][modus-themes]]
The colors looked a bit off the last time I tried it. I might have to spend some time tweaking the configs.

** General
#+begin_src elisp :tangle config.el
;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Athul Suresh"
      user-mail-address "athulsure@gmail.com")
#+end_src


#+begin_src elisp :tangle config.el
(setq display-line-numbers-type 't)
#+end_src

** Soft Line Wrapping
Getting soft line wraps to work was harder than I'd initially imagined. There wasn't an obvious solution in the inital google searches. Finally, I found =visual-fill-mode= in a comment which solved the problem.
#+begin_src elisp :tangle config.el
(setq visual-fill-column-width 80)
#+end_src
* Mac Specific Config
I was facing weird issues in the beginning because meta and super keys were bound to cmd key on my mac. This explicitly separates them out.
#+begin_src elisp :tangle config.el
(setq mac-option-key-is-meta nil
      mac-command-key-is-meta t
      mac-command-modifier 'meta
      mac-option-modifier 'super)
#+end_src
* Org-Mode
** General
I track all my notes in a private git repo. This was a practice I started when I using VSCode-Foam and it has carried over to Emacs.
#+begin_src elisp :tangle config.el
(setq org-directory "~/scaling-bassoon/")
#+end_src

I was absolutely fascinated by the concept of drawers and adding context to tasks. Logbook seems to be the supported way to do it.
#+begin_src elisp :tangle config.el
(setq org-log-into-drawer "LOGBOOK")
#+end_src

** Completions
I'm not a fan of the auto-complete that pops up while typing english in org mode. I would rather do without it.
#+begin_src elisp :tangle config.el
(after! org-mode
  (set-company-backend! 'org-mode nil))
#+end_src
* LSP
The LSP is the most amazing plugin I've used in a while, except for the weird part where the doc popup is annoying transparent. And it makes my cursor disapper sometimes. Disabling it until I can find a more permanent solution.
#+begin_src elisp :tangle config.el
(setq lsp-ui-doc-enable nil)
#+end_src

* Flycheck
I prefer a nice little indicator on the margin to denote lines having syntax errors. It's easier to parse with the naked eye.
#+begin_src elisp :tangle config.el
(setq-default left-fringe-width 1 right-fringe-width 8
              left-margin-width 1 right-margin-width 0)
(setq-default flycheck-indication-mode 'left-margin)
#+end_src

* Org Roam v2
#+begin_src elisp :tangle config.el
(setq org-roam-directory "~/scaling-bassoon/org-roam")
#+end_src

* Projectile
Set project paths for projectile
#+begin_src elisp :tangle config.el

(setq project-path '("~/src" "~/scaling-bassoon"))

(if (string= (system-name) mac-work)
    (cons "~/setu" project-path))

(setq projectile-project-search-path project-path)

#+end_src

I was facing issues where projectile was showing me stale data in the file search bufer (new files / deleted files). Disabling the cache seems to have fixed the issue.
#+begin_src elisp :tangle config.el
(setq projectile-enable-caching nil)
#+end_src

* Common Lisp
Lookup wasn't working as expected. Guess, I'll have to set those manually.
** TODO This might be shitty
#+begin_src elisp :tangle config.el
(set-lookup-handlers! 'lisp-mode
  :definition #'sly-describe-symbol
  :references #'sly-describe-symbol
  :documentation #'sly-describe-symbol)
#+end_src


config.el
(setq mac-work "Athuls-MacBook-Pro.local")
(setq mac-personal "MtDoom.local")

;; Doom exposes five (optional) variables for controlling fonts in Doom. Here
;; are the three important ones:
;;
;; + `doom-font'
;; + `doom-variable-pitch-font'
;; + `doom-big-font' -- used for `doom-big-font-mode'; use this for
;;   presentations or streaming.
;;
;; They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
;; font string. You generally only need these two:
(setq doom-font (font-spec :family "JetBrainsMono Nerd Font" :size 15))

;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:
(setq doom-theme 'doom-one-light)

;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Athul Suresh"
      user-mail-address "athulsure@gmail.com")

(setq display-line-numbers-type 't)

(setq visual-fill-column-width 80)

(setq mac-option-key-is-meta nil
      mac-command-key-is-meta t
      mac-command-modifier 'meta
      mac-option-modifier 'super)

(setq org-directory "~/scaling-bassoon/")

(setq org-log-into-drawer "LOGBOOK")

(after! org-mode
  (set-company-backend! 'org-mode nil))

(setq lsp-ui-doc-enable nil)

(setq-default left-fringe-width 1 right-fringe-width 8
              left-margin-width 1 right-margin-width 0)
(setq-default flycheck-indication-mode 'left-margin)

(setq org-roam-directory "~/scaling-bassoon/org-roam")

(setq project-path '("~/src" "~/scaling-bassoon"))

(if (string= (system-name) mac-work)
    (cons "~/setu" project-path))

(setq projectile-project-search-path project-path)

(setq projectile-enable-caching nil)

(set-lookup-handlers! 'lisp-mode
  :definition #'sly-describe-symbol
  :references #'sly-describe-symbol
  :documentation #'sly-describe-symbol)
